content/ui-patterns/forms.mdx

```diff
@@ -0,0 +1,382 @@
+    <img
+        width="456"
+        alt="checkbox group with an error"
+        src="https://user-images.githubusercontent.com/2313998/164065224-d1ba56ce-c632-4318-b9b2-b75940f771a6.png"
+    />
+    <Box as="p" mt="0">A validation message may be used to describe an error for a group of form controls. For example: if a user checks 3 checkboxes when they’re only allowed to check 2 options.</Box>
+</Box>
+
+## Submission
+
+Forms should follow consistent patterns for [submitting and saving data](/ui-patterns/saving).
+
+## Related components
+
+<Box mb={4} display="flex" alignItems="flex-start" flexDirection={["column", "column", "column", "column", "row"]} sx={{gap: 3}}>
+  <img
+    width="456"
+    role="presentation"
+    src="https://user-images.githubusercontent.com/293280/123881067-cdbaea00-d8f8-11eb-98e4-e57c64489308.png"
+  />
+  <div>
+    <Heading fontSize={3}>Action menu with selection</Heading>
+    <p>If the parent form control provides sufficient visual context, you may visually hide the label. However, you must specify text for a visually hidden label that is accessible to screen readers.</p>
+    <a href="https://primer.style/react/ActionMenu#with-selection">Primer React implementation</a>
+  </div>
+</Box>
+
+<Box mb={4} display="flex" alignItems="flex-start" flexDirection={["column", "column", "column", "column", "row"]} sx={{gap: 3}}>
+  <img
+    role="presentation"
+    width="456"
+    src="https://user-images.githubusercontent.com/2313998/168104391-d83bef5e-a628-4788-b1c2-6d72980fa508.png"
+  />
+  <div>
+    <Heading fontSize={3}>Autocomplete</Heading>
+    <p>
+      An autocomplete input renders a text input that allows a user to quickly filter through a list of options to pick one or more values.
+    </p>
+      <a href="https://primer.style/react/Autocomplete">Primer React implementation</a>
+  </div>
+</Box>
+
+<Box mb={4} display="flex" alignItems="flex-start" flexDirection={["column", "column", "column", "column", "row"]} sx={{gap: 3}}>
+  <img
+    role="presentation"
+    width="456"
+    src="https://user-images.githubusercontent.com/2313998/166846801-91a03e2d-aedb-4ba3-b8ea-c7abcb7eadd6.png"
+  />
+  <div>
+    <Heading fontSize={3}>Checkbox group</Heading>
+    <p>
+      A set of checkboxes to let users make one or more selections from a short list of options
+    </p>
+      <a href="https://primer.style/react/CheckboxGroup">Primer React implementation</a>
+  </div>
+</Box>
+
+<Box mb={4} display="flex" alignItems="flex-start" flexDirection={["column", "column", "column", "column", "row"]} sx={{gap: 3}}>
+  <img
+    role="presentation"
+    width="456"
+    src="https://user-images.githubusercontent.com/2313998/166846802-7d3a0563-d853-4d7b-b26f-0bb4e8c0d3ad.png"
+  />
+  <div>
+    <Heading fontSize={3}>Form control</Heading>
+    <p>
+      A form control renders a labelled input and, optionally, associated validation text and/or hint text.
```

Comment:
This is probably out of scope for this particular task, but I just wanted to raise that if I wanted to have a group of radio inputs, I could pick either `Form control` or `Radio group` 😬 It's not totally clear why I should pick one vs. the other

Reply:
I will make sure this is clear when we write the guidelines for those components.

Incase you're curious: the main difference is that a `Form control` is used for a single input, and a `Radio group` is used to group multiple related radio inputs. The markup is rendered differently to support assistive technology for screen readers, and the typographic hierarchy is different. Compare this [RadioGroup example](https://primer.style/react/RadioGroup#with-caption) to this [FormControl example](https://primer.style/react/FormControl#with-a-caption).